File: src/ServerPulse.Frontend/src/app/modules/analyzer/domain/dtos/responses/statistics/serverLoadStatisticsResponse.ts
import { BaseStatisticsResponse, LoadEventResponse, LoadMethodStatisticsResponse, mapBaseStatisticsResponseToBaseStatistics, mapLoadEventResponseToLoadEvent, mapLoadMethodStatisticsResponseToLoadMethodStatistics, ServerLoadStatistics } from "../../../..";

export interface ServerLoadStatisticsResponse extends BaseStatisticsResponse {
    amountOfEvents: number;
    lastEvent: LoadEventResponse | null;
    loadMethodStatistics: LoadMethodStatisticsResponse | null;
}

export function mapServerLoadStatisticsResponseToServerLoadStatistics(response: ServerLoadStatisticsResponse): ServerLoadStatistics {
    return {
        ...mapBaseStatisticsResponseToBaseStatistics(response),
        amountOfEvents: response?.amountOfEvents,
        lastEvent: response.lastEvent === null ? null : mapLoadEventResponseToLoadEvent(response.lastEvent),
        loadMethodStatistics: response.loadMethodStatistics === null ? null : mapLoadMethodStatisticsResponseToLoadMethodStatistics(response.loadMethodStatistics)
    };
}

export function getDefaultLoadStatisticsResponse(): ServerLoadStatisticsResponse {
    return {
        id: "",
        collectedDateUTC: new Date(),
        amountOfEvents: 0,
        lastEvent: null,
        loadMethodStatistics: null
    };
}

File: src/ServerPulse.Backend/AnalyzerApi.Core/Dtos/Responses/Statistics/ServerLoadStatisticsResponse.cs
using AnalyzerApi.Core.Dtos.Responses.Events;

namespace AnalyzerApi.Core.Dtos.Responses.Statistics
{
    public class ServerLoadStatisticsResponse : BaseStatisticsResponse
    {
        public int AmountOfEvents { get; set; }
        public LoadEventResponse? LastEvent { get; set; }
        public LoadMethodStatisticsResponse? LoadMethodStatistics { get; set; }
    }
}